.github/workflows/publish_after_push.yml

```
@@ -4,41 +4,53 @@ on:
   workflow_dispatch:
   push:
     branches: ["gh-pages"]
+  pull_request:
+    branches: ["gh-pages"]
 
 permissions:
   contents: write
   pages: write
   id-token: write
 
 jobs:
-  gh-pages-on-push:
+  lint-and-test-and-build:
     runs-on: ubuntu-latest
-    environment:
-      name: github-pages
-      url: ${{ steps.deployment.outputs.page_url }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-        with:
-          ref: "gh-pages"
 
       - name: Setup Node
         uses: actions/setup-node@v4
         with:
           node-version: "22"
 
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Lint
+        run: npm run lint
+
+      - name: Test
+        run: npm test
+
```

Actually, `npm ci` is actually not "npm install that you run in ci", it's "npm **c**lean **I**nstall" which is sometimes what you need but quite often it's not. Here regular `npm install` would just be fine.

We need to decide whether we want to use make or npm for tasks. We have a weird split now where things are defined in two places, and it makes us run things too often (`make build` at line 37 also does npm install). 
I'm partial to make because JSON strings are not great for longer build scripts. I'd prefer to remove scripts from the package.json completely and put everything in the Makefile:

```suggestion
      - name: Install dependencies
        run: make build

      - name: Lint
        run: make link

      - name: Test
        run: make test
```

---

“ci” is not wrong but since we start from a clean checkout it doesn’t do anything extra useful, especially since we run install afterwards anyway, potentially in theory modifying the package lock because now we run install based on the package json instead of the lock file… or I dunno, maybe that’s not true but I don’t want to have to think about npm internals. If we run ci let’s just run ci once and then not touch it again.

So ideally we do resolve this neatly. If running make on Windows is a problem (what do you use for shell scripts and git then? Doesn’t that come with make? Make is in POSIX IIRC so it should be quite portable)… we can put the downloading bits in a shell script and everything else in the package.json?